src/components/views/EditMilestone.jsx

```diff
@@ -128,13 +143,31 @@ class EditMilestone extends Component {
               );
             });
         } else {
+          const qs = queryString.parse(this.props.location.search);
           CampaignService.get(this.props.match.params.id)
             .then(campaign => {
               if (campaign.projectId < 0) {
                 this.props.history.goBack();
               } else {
                 const { milestone } = this.state;
-                milestone.recipientAddress = this.props.currentUser.address;
+                validQueryStringVariables.forEach(variable => {
+                  if (qs[variable]) {
+                    if (variable === 'fiatAmount') {
+                      this.setMaxAmount('', qs[variable]);
```

Comment:
All these functions are doing setState. This is dangerous and should be avoided. I think it would be best to set the milestone object yourself and if needed at the end of this componentWillMount methods (but I don't think you need to)